Swiftx/Combinators.swift

```diff
@@ -88,10 +88,17 @@ public func |> <A, B>(a : A, f : A -> B) -> B {
 /// point at which further application of x to a function is the same x.
 ///
 ///     x = f(x)
-public func fix<A>(f : ((A -> A) -> A -> A)) -> A -> A {
+public func fix<A, B>(f : (A -> B) -> A -> B) -> A -> B {
 	return { x in f(fix(f))(x) }
 }
 
+/// The fixpoint (or Y) combinator computes the least fixed point of an equation. That is, the first
+/// point at which further application of x to a function is the same x.
+/// fixt is the exception-enabled version of fix.
+///
```

Comment:
good to know markdown is partially supported :)

On Sun, Sep 6, 2015 at 9:18 PM Robert Widmann <notifications@github.com>
wrote:

> In Swiftx/Combinators.swift
> <https://github.com/typelift/Swiftx/pull/30#discussion_r38825650>:
>
> >  	return { x in f(fix(f))(x) }
> >  }
> >
> > +/// The fixpoint (or Y) combinator computes the least fixed point of an equation. That is, the first
> > +/// point at which further application of x to a function is the same x.
> > +/// fixt is the exception-enabled version of fix.
> > +///
>
> Whoops, other way around:
>
> /// The fixpoint (or Y) combinator computes the least fixed point of an equation. That is, the first
>
> /// point at which further application of x to a function is the same x.
>
> ////// `fixt` is the exception-enabled version of fix.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/typelift/Swiftx/pull/30/files#r38825650>.
>